### PR TITLE
Test for dnf-automatic reporting a transaction error

### PR DESCRIPTION
**Pull request by jan-kolarik**
Requires https://github.com/rpm-software-management/dnf/pull/1923.

### CODE REVIEW

File: dnf-behave-tests/dnf/dnf-automatic/error-report.feature [hunk 1]
```diff
@@ -0,0 +1,25 @@
+@no_installroot
+Feature: dnf-automatic reports an error when transaction failed
+
+
+Background:
+Given I delete file "/etc/yum.repos.d/*.repo" with globs
+  And I create file "/etc/dnf/dnf.conf" with
```
Comment (j-mracek) [inline dnf-behave-tests/dnf/dnf-automatic/error-report.feature:7]: What about to add a comment why the scenario requires plugins to be switched off? I know that this is dnf test default but this is not a reason.

File: dnf-behave-tests/dnf/dnf-automatic/error-report.feature [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    [main]
+    plugins=0
+    """
+
+
+@bz2170093
+Scenario: dnf-automatic reports an error when package installation failed
+  Given I use repository "dnf-ci-automatic-update"
+    And I successfully execute dnf with args "install test-1.0"
+   When I execute dnf-automatic with args "--installupdates"
```
Comment (j-mracek) [inline dnf-behave-tests/dnf/dnf-automatic/error-report.feature:18]: From test description it is not clear why the command is supposed to fail. What about to add a comment that the step will try  to install test-1-1 with broken scriptlet. The expected result is clear from test but not why it is expected.

File: dnf-behave-tests/dnf/dnf-automatic/error-report.feature (hunk 1 above)
Comment (jan-kolarik) [reply dnf-behave-tests/dnf/dnf-automatic/error-report.feature:7]: This was actually copied from the other existing `dnf-automatic` test, but it is not even needed by this test. I can drop it.

**Comment by j-mracek** [comment]
LGTM

**Comment by j-mracek** [comment]
It looks like that plugin disablement is required for outpu test:
```
       Error in PREIN scriptlet in rpm package test  |  (7, "Failed to connect to 127.0.0.1 port 9050 after 0 ms: Couldn't connect to server")
      Error: Transaction failed                     |  Disabling torproxy plugin: cannot connect to the Tor network
                                                    |  Errors during downloading metadata for repository '_dnf_local':
                                                    |    - Curl error (37): Couldn't read a file:// file for file:///var/lib/dnf/plugins/local/repodata/repomd.xml [Couldn't open file /var/lib/dnf/plugins/local/repodata/repomd.xml]
                                                    |  Error: Failed to download metadata for repo '_dnf_local': Cannot download repomd.xml: Cannot download repodata/repomd.xml: All mirrors were tried
                                                    |  Ignoring repositories: _dnf_local
                                                    |  snapper: connect to snapperd failed: org.freedesktop.DBus.Error.FileNotFound: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
                                                    |  Error in PREIN scriptlet in rpm package test
                                                    |  No installed rpm found for package test-1.1-1.x86_64, will not sync SWID tag.
                                                    |  Error: Transaction failed
```

**Comment by jan-kolarik** [comment]
Hmm, yes... I was testing it just locally running the single test and didn't realize that.

**Comment by j-mracek** [comment]
LGTM

**Pull request merged**